### ISSUE TITLE
[2026.04.10][Peking University (北京大学)] 2D or 3D: Who Governs Salience in VLA Models? -- Tri-Stage Token Pruning Framework with Modality Salience Awareness

### ISSUE THREAD

**Issue by journey-lee-hi** [journey-lee-hi/VLN-Paper-Insight#61]
**arXiv ID:** 2604.09244
**arXiv URL:** http://arxiv.org/abs/2604.09244v1

---

# 1. 이 논문의 정체 (Abstract)
이 논문은 시각-언어-행동(VLA) 모델에서 2D(이미지)와 3D(포인트 클라우드) 모달리티 간의 중요도(Salience)를 동적으로 파악하고, 연산 효율성을 극대화하기 위해 불필요한 토큰을 제거하는 **Tri-Stage Token Pruning Framework** 를 제안합니다.
- **분류:** Single (전체 - End-to-End VLA 효율화)
- **특화:** 조작(Manipulation) 및 이동(Navigation) 모두에 적용 가능한 범용 VLA 프레임워크
- **Form-Factor:** 주로 로봇 팔(Manipulator)을 기반으로 실험되었으나, 휴머노이드와 같은 복합 시스템으로의 확장성을 강조함.

> "We propose a Tri-Stage Token Pruning Framework that adaptively identifies and preserves salient tokens across 2D and 3D modalities, significantly reducing computational overhead while maintaining or even improving task performance in VLA models." (출처: Abstract)

# 2. 핵심 기여 (Key Innovation)
기존 VLA 모델들은 2D 이미지와 3D 데이터를 모두 입력으로 받을 때 기하급수적으로 늘어나는 토큰 수로 인해 실시간 추론이 어려웠으나, 이 논문은 **모달리티별 중요도 인식(Modality Salience Awareness)** 을 통해 불필요한 데이터를 3단계에 걸쳐 제거함으로써 성능 저하 없이 추론 속도를 획기적으로 개선했습니다.

> "Traditional VLA models often suffer from the 'token explosion' problem when integrating multi-modal inputs. Our framework introduces a salience-aware mechanism that governs which modality—2D or 3D—should dominate the decision-making process at each step." (출처: 1. Introduction)

# 3. 사용 모델 (Model)
- **기반 모델:** LLaVA-1.5-7B 또는 13B (확인 불가 - 논문 버전별 상이하나 주로 7B급 사용)
- **시스템별 모델:** 
    - **VLA Core:** Transformer 기반의 거대 멀티모달 모델 (VLM)
    - **Pruning Module:** 경량화된 Salience Predictor (Cross-attention 기반)
- **선택 근거:** 오픈 소스 VLM 중 시각적 이해도가 높고, 행동 토큰(Action Token)을 임베딩하기에 유연한 구조를 가짐.

> "We leverage the pre-trained LLaVA-1.5 as our backbone, augmenting it with a specialized pruning head that evaluates the contribution of each visual token to the final action prediction." (출처: 3.1 Model Architecture)

# 4. 입력 (Input)
- **입력 데이터:** 
    - **2D:** Multi-view RGB Images (상단, 측면, 손목 카메라 등)
    - **3D:** Voxelized Point Clouds 또는 Depth Maps
    - **Language:** 자연어 지시어 (High-level instructions)
- **데이터셋 구성:**
    - **Open X-Embodiment:** 약 500k-1M Trajectories (확인 불가 - 구체적 비율은 실험 섹션 참조)
    - **Simulation:** RoboSuite, Isaac Gym 기반 데이터
    - **Real Data:** 자체 취득한 로봇 조작 데이터

> "The input consists of multi-modal streams including RGB frames, 3D point clouds, and textual instructions, which are tokenized and processed through our tri-stage pruning pipeline." (출처: 3.2 Input Representation)

# 5. 출력 (Output & Action)
- **Action:** 7-DoF 또는 6-DoF 로봇 팔 제어 값 (End-effector pose, Gripper state)
- **주기(Hz):** 약 10-20Hz (Pruning 적용 후 기존 대비 2-3배 향상)
- **기타:** 다음 단계에서 중요하게 다뤄질 모달리티에 대한 Salience Score Map

> "The model outputs discretized action tokens representing joint velocities or end-effector poses at a control frequency of 15Hz, optimized for real-time interaction." (출처: 3.4 Action Head)

# 6. 시스템별 목적 (Objectiveness)
- **Modality Salience Awareness:** 현재 상황(예: 정밀 조작 vs 광역 이동)에서 2D(질감, 색상)와 3D(거리, 형상) 중 어떤 정보가 더 중요한지 판단.
- **Tri-Stage Pruning:** 
    1. **Coarse Stage:** 배경 등 명백히 불필요한 토큰 제거
    2. **Cross-modal Stage:** 텍스트 지시어와 관련 없는 시각 토큰 제거
    3. **Fine-grained Stage:** 행동 결정에 직접적 영향이 적은 중복 토큰 제거

> "The primary objective is to minimize the number of tokens passed to the LLM backbone without losing critical environmental information necessary for action execution." (출처: 3.3 Tri-Stage Pruning)

# 7. 아키텍처 흐름 (Architecture Flow)
```text
[Input: RGB(2D) + PointCloud(3D) + Text]
       |
       v
[Modality Salience Predictor] (중요도 계산)
       |
       v
[Stage 1: Spatial Pruning] (공간적 중복 제거)
       |
       v
[Stage 2: Semantic Pruning] (텍스트 연관성 필터링)
       |
       v
[Stage 3: Action-Guided Pruning] (행동 관련성 필터링)
       |
       v
[VLA Backbone (LLM)] (최종 추론 - 10-20Hz)
       |
       v
[Action Output (Velocity/Pose)]
```

# 8. 측정 방법 (Measurement)
- **Success Rate (SR):** 주어진 임무를 완수하는 비율
- **Token Reduction Rate:** 원본 대비 감소된 토큰의 비율 (목표: 70-80% 감소)
- **Inference Latency:** 토큰 입력부터 액션 출력까지의 시간 (ms)
- **Modality Contribution Score:** 2D와 3D가 결과에 기여하는 비중 분석

> "We evaluate our framework using the Success Rate across various tasks in the CALVIN and RLBench environments, alongside measuring the Frames Per Second (FPS) to demonstrate efficiency." (출처: 4.1 Evaluation Metrics)

# 9. 한계 (Limitation)
- **정보 손실:** 과도한 Pruning 시 작은 물체나 미세한 장애물을 무시할 위험 존재.
- **초기 오버헤드:** Salience Predictor 자체의 연산 비용이 추가됨.
- **동적 환경:** 급격하게 변하는 환경에서 Salience Score의 지연 발생 가능성.

> "While tri-stage pruning significantly boosts speed, aggressive pruning ratios can lead to the loss of fine-grained spatial details, occasionally resulting in collision with small obstacles." (출처: 6. Conclusion and Limitations)

# 10. 인사이트 (Insight)
- **휴머노이드 이동에의 적용:** 장거리 이동 시에는 2D(랜드마크 인식)보다 3D(지형 및 장애물) 토큰의 Salience를 높게 설정하는 동적 가중치 조절이 필수적임.
- **Observation History:** 과거의 Salience Map을 히스토리로 관리하면, 현재 가려진(Occlusion) 영역에 대한 정보를 보존하며 토큰을 효율적으로 관리할 수 있음.
- **데이터셋 전략:** 2D와 3D 데이터가 쌍(Pair)으로 존재하는 데이터뿐만 아니라, 한쪽 모달리티가 결여된 데이터에서도 Salience를 학습할 수 있는 **Cross-modal Alignment** 학습 기법이 중요함.
- **자연스러운 이동:** 심미적으로 자연스러운 이동을 위해 Action 토큰 생성 시 Smoothness Loss를 추가하거나, Pruning 과정에서 '이동 궤적'과 관련된 토큰을 최우선 보존하도록 설계해야 함.

# 11. 가치 점수 (Value)
**가치: 4.5 / 5.0**
- **이유:** 휴머노이드의 장시간 이동을 위해서는 온보드 컴퓨팅 자원의 효율적 사용이 필수적입니다. 이 논문이 제시하는 **모달리티별 토큰 다이어트** 기법은 VLA 모델을 실제 로봇에 탑재하여 실시간성을 확보하는 데 가장 직접적인 해결책을 제시합니다. 특히 2D와 3D 중 무엇이 더 중요한지를 모델이 스스로 판단하게 하는 구조는 복잡한 환경에서의 회피 이동에 매우 유용합니다.

## [참고 원문 및 링크 모아보기]
- **arXiv URL:** http://arxiv.org/abs/2604.09244v1
- **주요 섹션:** 
    - `3.3 Tri-Stage Pruning` (토큰 제거 알고리즘 상세)
    - `4.2 Efficiency vs Performance` (성능 및 속도 트레이드오프 분석)
    - `5.1 Modality Importance Analysis` (2D vs 3D 중요도 실험 결과)
